sdk/src/main/java/com/dailymotion/android/player/sdk/VisitorInfoManager.kt

```diff
@@ -0,0 +1,27 @@
+package com.dailymotion.android.player.sdk
+
+import android.content.Context
+import com.google.android.gms.ads.identifier.AdvertisingIdClient
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import kotlinx.coroutines.withContext
+
+class VisitorInfoManager {
+
+    private lateinit var advertisingInfo: AdvertisingIdClient.Info
+    private val mutex = Mutex()
+
+    suspend fun getAdvertisingInfo(context: Context): AdvertisingIdClient.Info = mutex.withLock {
+        if (::advertisingInfo.isInitialized.not() || advertisingInfo.id.isNullOrBlank()) {
+            advertisingInfo = withContext(Dispatchers.IO) {
+                try {
+                    AdvertisingIdClient.getAdvertisingIdInfo(context)
+                } catch (e: Exception) {
+                    AdvertisingIdClient.Info("", false)
```

Comment:
In order to keep the same behaviour as the previous implementation, can you:
1. log the exception caught
2. return a null object if there is an error. I would avoid creating ourself the object since the latest google documentation deprecated the constructor: https://developers.google.com/android/reference/com/google/android/gms/ads/identifier/AdvertisingIdClient.Info#public-info-string-advertisingid,-boolean-limitadtrackingenabled

Reply:
OK 👍